### PR TITLE
[Modular] Nanotrasen-Brand Meals Ready to Eat™

### PR DESCRIPTION
**Pull request by ShadeAware**
## About The Pull Request
Adds new, modern types of MREs, along with a vendor to gather each! Each MRE comes with a randomized main dish, along with some side sacks, condiment packets, a fork and some water.

## Why It's Good For The Game
The MREs provide hungry crewmembers with ways to get some much-needed sustenance when the chef and/or botany are either out, or not doing their job. Lets be honest, its usually both.

## Changelog
:cl:
add: New MRE Packets, sprites taken from Baystation
add: MRE-Vend, for the new MRE Packets. Can be found near the kitchen, or bar depending on the map.
/:cl:

### CODE REVIEW

File: modular_skyrat/code/game/objects/items/storage/MREs.dm [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		/obj/item/reagent_containers/food/condiment/pack/))
+	STR.can_hold = can_hold
+
+/obj/item/storage/mre/PopulateContents()
+	new /obj/item/reagent_containers/food/snacks/soup/mre(src)
+	switch (rand(1,2))
+		if(1)
+			new /obj/item/reagent_containers/food/snacks/sosjerky/healthy/mre(src)
+		if(2)
+			new /obj/item/reagent_containers/food/snacks/no_raisin/healthy/mre(src)
```
Comment (jjpark-kb) [inline modular_skyrat/code/game/objects/items/storage/MREs.dm:23]: use pick() instead

File: modular_skyrat/code/game/objects/items/storage/MREs.dm [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		if(1)
+			new /obj/item/reagent_containers/food/snacks/meatcube(src)
+		if(2)
+			new /obj/item/reagent_containers/food/snacks/bbqribs(src)
+		if(3)
+			new /obj/item/reagent_containers/food/snacks/cubancarp(src)
+		if(4)
+			new /obj/item/reagent_containers/food/snacks/taco/plain(src)
+		if(5)
+			new /obj/item/reagent_containers/food/snacks/burger/plain(src)
```
Comment (jjpark-kb) [inline modular_skyrat/code/game/objects/items/storage/MREs.dm:43]: same here

File: modular_skyrat/code/game/objects/items/storage/MREs.dm [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		if(1)
+			new /obj/item/reagent_containers/food/snacks/salad/herbsalad(src)
+		if(2)
+			new /obj/item/reagent_containers/food/snacks/salad/citrusdelight(src)
+		if(3)
+			new /obj/item/reagent_containers/food/snacks/loadedbakedpotato(src)
+		if(4)
+			new /obj/item/reagent_containers/food/snacks/carrotfries(src)
+		if(5)
+			new /obj/item/reagent_containers/food/snacks/salad/boiledrice(src)
```
Comment (jjpark-kb) [inline modular_skyrat/code/game/objects/items/storage/MREs.dm:66]: same here

File: modular_skyrat/code/game/objects/items/storage/MREs.dm [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		if(2)
+			new /obj/item/reagent_containers/food/snacks/stuffedlegion(src)
+		if(3)
+			new /obj/item/toy/crayon/rainbow/big(src)
+		if(4)
+			new /obj/item/reagent_containers/food/snacks/burger/corgi(src)
+		if(5)
+			new /obj/item/reagent_containers/food/snacks/burger/roburger(src)
+		if(6)
+			new /obj/item/reagent_containers/food/snacks/snowcones/rainbow(src)
```
Comment (jjpark-kb) [inline modular_skyrat/code/game/objects/items/storage/MREs.dm:92]: same here

File: modular_skyrat/code/game/objects/items/storage/MREs.dm (hunk 1 above)
Comment (ShadeAware) [reply modular_skyrat/code/game/objects/items/storage/MREs.dm:23]: tried pick() and it didn't work, I showed you this earlier.

File: modular_skyrat/code/game/objects/items/storage/MREs.dm (hunk 2 above)
Comment (ShadeAware) [reply modular_skyrat/code/game/objects/items/storage/MREs.dm:43]: tried pick() and it didn't work, I showed you this earlier.

File: modular_skyrat/code/game/objects/items/storage/MREs.dm (hunk 4 above)
Comment (ShadeAware) [reply modular_skyrat/code/game/objects/items/storage/MREs.dm:92]: tried pick() and it didn't work, I showed you this earlier.

File: modular_skyrat/code/game/objects/items/storage/MREs.dm (hunk 3 above)
Comment (ShadeAware) [reply modular_skyrat/code/game/objects/items/storage/MREs.dm:66]: tried pick() and it didn't work, I showed you this earlier.

**Comment by davidchan13** [comment]
The MREs provide hungry crewmembers with ways to get some much-needed sustenance when ~~the chef and/or botany are either out, or not doing their job~~ fully stocked science department completely fails to upgrade and support the service department 90 minutes into the shift despite having an abundance of resources.

FTFY

**Comment by Owai-Seek** [comment]
I don't really like the idea of making chef an even more useless role, being able to literally purchase somewhat decent food from a vendor without interacting with anyone. With there being no food available, it encourages people to fill the role of chef, or at the very least, -try- to fill it, or interact with cargo.

Changing the vending machine to be its own machine instead of a specific map "Condivend"  would be nice, but there's already a (barely used) vendor for bare-minimum tier food. (Sustenance Vendor)

**Comment by ShadeAware** [comment]
Maybe when Chef or Cargo ever start to bother to get off their ass and purchase food that everyone can actually eat, and not just food for Omnivores. Besides, you cant get the refill for the vendor, so the supply is limited.

**Comment by Owai-Seek** [comment]
There are plenty of options available if no chef is available, without giving people more access to easy-grab food. I don't mind the MRE's themselves if they're added to random-spawn, but starting out every shift with twenty full food packs for people to just grab at shift start is stupid. People will grab MRE's with their shift-start money and never leave their department. There will be no conflict or interaction over a lack of food if they never have to go to the kitchen.

There are several viable options if no food is available. You can make food in the public garden, xenobio can make food with extracts, you can temporarily tide yourself over with junkfood from a snack vendor, or order things from cargo.

You shouldn't be able to get everything done without interacting with other departments, while it can be irritating that the chef or botany hasn't made something for people to eat, you can easily ask them to, and a lot of the time they'll do it. If they don't, well, check with cargo or tell the HoP that these people aren't doing their jobs. Maybe ask whoever's there if they need help and teach a new player how to cook, or hop the table and whip up some eggs, cheese, or tofu.

**Comment by ShadeAware** [comment]
There doesn't need to be conflict over absolutely fucking everything. If Chefs don't start actually making food everyone can eat, then tough shit. They can cry and bitch and moan about MREs all they want.

**Comment by XyvMeph** [comment]
As a service main that frequently works with chefs, and knowing a lot of chefs, I can personally guarantee you that, diets aside, let alone Shiva's fucking diet, can be worked around. Vendors already have food all across the station exclusively for circumventing this.

This literally just sounds like a PR that is exclusively here for the sole purpose of invalidating chefs. If people want MRE's so dam badly they can go hunting in maintenance for them, it's literally not hard.

Chef is a service job, and, frankly, rather, arguably, is actually the most important one. Botany doesn't break the station if it isn't there. Bartender sure as shit doesn't even so much as bop the station if they aren't present. Chef, however, is the job that keeps space men from running slow, and I have literally never seen the HoP, a job that honestly should require chef hours (because service) run in there to pick up the slack like a restaraunt manager.

But that's all useless jargon.

Laymen's terms: For fuck's sake don't just completely invalidate an entire job just because there's a few shifts where we either don't have one, or they're some useless husk of a character that's walking around shitting it up with their other husk of a character friends. The chance that Cargo *and* Chef are both not available to get food for the whole crew is a lot slimmer than you think.

Look at it this way.

Least you're not a lizard starving for 3 whole hours.

**Comment by jjpark-kb** [comment]
Its been said before: This PR may invalidate chef.

**Pull request closed without merging**